### PR TITLE
feature: add option to pass EnableRemoteWriteReceiver to Prometheus CR

### PR DESCRIPTION
**Pull request by jan--f**
Fixes: https://github.com/rhobs/observability-operator/issues/160

Signed-off-by: Jan Fajerski <jfajersk@redhat.com>

### CODE REVIEW

File: pkg/apis/monitoring/v1alpha1/types.go [hunk 1]
```diff
@@ -96,6 +96,9 @@ type PrometheusConfig struct {
 	// Define ExternalLabels for prometheus
 	// +optional
 	ExternalLabels map[string]string `json:"externalLabels,omitempty"`
+	// Enable Prometheus to be used as a receiver for the Prometheus remote write protocol. Defaults to the value of `false`.
+	// +optional
+	EnableRemoteWriteReceiver bool `json:"enableRemoteWriteReceiver,omitempty"`
```
Comment (sthaha) [inline pkg/apis/monitoring/v1alpha1/types.go:101]: shouldn't this be `*bool` so that `false` isn't omitted?

Comment (jan--f) [reply pkg/apis/monitoring/v1alpha1/types.go:101]: Iiuc a pointer allows us to distinguish unset from `false`/empty. I don't think we need this here, do we?

Comment (jan--f) [reply pkg/apis/monitoring/v1alpha1/types.go:101]: This is also consistent with [prometheus-operator's spec](https://github.com/prometheus-operator/prometheus-operator/blob/50ec1f57f19289cda613d484f3e66327bfeb3328/pkg/apis/monitoring/v1/types.go#L169).

Comment (sthaha) [reply pkg/apis/monitoring/v1alpha1/types.go:101]: I tested this against kind and it actually works as expected. My understanding was that `bool` value of false is the zero value and if you applied the following 

```yaml
spec:
  prometheusConfig:
    enableRemoteWriteReceiver: false
```
The zero value of false will be omitted (just like json.Marshal does) and `kubectl get -o yaml monitoringstack foobar` will **not** print `enableRemoteWriteReceiver: false` but to the contrary, it does in fact print it. I think json marshalling should have been changed to accomodate this behaviour.

**Review by sthaha — APPROVED** [review]

**Pull request merged**